plugins/catalog-backend/src/processing/ProcessorOutputCollector.ts

```diff
@@ -89,6 +90,17 @@ export class ProcessorOutputCollector {
         return;
       }
 
+      // The processor contract says you should return the "trunk" (current)
+      // entity, not emit it. But it happens that this is misunderstood or
+      // accidentally forgotten. This can lead to circular references which at
+      // best is wasteful, so we try to be helpful by ignoring such emitted
+      // entities.
+      if (
+        stringifyEntityRef(entity) === stringifyEntityRef(this.parentEntity)
+      ) {
+        return;
```

Comment:
Just as an option, should we be throwing instead? Definitely see that this can be a bit less impactful to ship though

Reply:
Yeah exactly, it felt like it'd possibly cause a bit too much commotion and not providing enough actual value / avoiding actual risk, to be worth it

Reply:
I think it would make sense to log a warning, or maybe even error level?

Reply:
Ah yeah that I can get behind. At this point we don't know which the offending processor is though 🤔 I'll look into it.

Reply:
K, updated with some more proper (hopefully) logging.